dragonfly/opt/unittest_euclidean_random_optimiser.py

```diff
@@ -165,7 +193,29 @@ def run_optimiser(cls, func_caller, worker_manager, max_capital, mode, *args, **
     """ Runs optimiser. """
     return random_optimiser.random_optimiser_from_func_caller(
              func_caller, worker_manager, max_capital, mode)
-
+  
+  def test_ask_tell(self):
+    """ Testing random optimiser with ask tell interface. """
+    self.report('Testing %s using the ask-tell interface.'%(type(self)))
+    domain = domains.EuclideanDomain([[0, 2.3], [3.4, 8.9], [0.12, 1.0]])
+    func_caller = EuclideanFunctionCaller(None, domain)
+    opt = random_optimiser.EuclideanRandomOptimiser(func_caller, ask_tell_mode=True)
+    opt.initialise()
+
+    def evaluate(x):
+      return get_syn_func_caller('hartmann3', noise_type='gauss', noise_scale=0.1).func(x)
+
+    best_x, best_y = None, float('-inf')
+    for _ in range(20):
+      x = opt.ask()
+      y = evaluate(x)
+      opt.tell([(x, y)])
+      self.report('x: %s, y: %s'%(x, y))
+      if y > best_y:
+        best_x, best_y = x, y
+    self.report("-----------------------------------------------------")
+    self.report("Optimal Value: %s, Optimal Point: %s"%(best_y, best_x))
+    
```

Comment:
@anthonyhsyu I meant can you run the optimiser in ask-tell mode vs regular mode and see if the answers are consistent for the same test function - they shouldn't be the same but should be close. If BO is too expensive, you can try random search or evolutionary algorithms.